lib/elixir/lib/macro.ex

```diff
@@ -1121,6 +1130,25 @@ defmodule Macro do
     :error
   end
 
+  defp parse_sigil("sigil_" <> <<letter>>) when letter >= ?a and letter <= ?z do
+    {:downcase, <<letter>>}
+  end
+
+  defp parse_sigil("sigil_" <> letters) do
```

Comment:
I think we could check if the first letter is an uppercase letter:

```elixir
iex> ast = {:sigil_foo, [delimiter: "<"], [{:<<>>, [], ["hello"]}, []]}
iex> Macro.to_string(ast)
"~foo<hello>"
```

Reply:
Good call!